enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/jwt/InternalAPIKeyAuthenticator.java

```diff
@@ -98,6 +98,8 @@ public AuthenticationContext authenticate(RequestContext requestContext) throws
                 String internalKey = extractInternalKey(requestContext);
                 // Remove internal key from outbound request
                 requestContext.getRemoveHeaders().add(securityParam);
+                // Avoid internal key being published to the Traffic Manager
+                requestContext.getProtectedHeaders().add(securityParam);
```

Comment:
both should be added outside of the auth filters imho

Reply:
fixed with https://github.com/wso2/product-microgateway/pull/2563/commits/9ca6828292e1cc968c8af8031e58816a1b85a49a